### PR TITLE
fix broken build

### PR DESCRIPTION
**Pull request by dcrissman**
Not sure how/why the master build broke, but this fixes it (and a little bit of cleanup).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -147,12 +147,23 @@
...
+            </dependency>
+            <dependency>
                 <groupId>com.redhat.lightblue</groupId>
                 <artifactId>lightblue-core-test</artifactId>
                 <version>1.5.0-SNAPSHOT</version>
                 <scope>test</scope>
             </dependency>
             <dependency>
+                <groupId>com.redhat.lightblue.mongo</groupId>
+                <artifactId>lightblue-mongo-test</artifactId>
```
Comment (jewzaam) [inline pom.xml]: An undocumented pattern we've been following is to only put dependencies outside of the repo in dependencyManagement.  In my mind it's defining an artifact that doesn't exist yet doing this, but it technically works.

File: hystrix/pom.xml [hunk 2]
```diff
@@ -24,16 +24,17 @@
         <artifactId>lightblue-mongo-pom</artifactId>
         <version>1.5.0-SNAPSHOT</version>
     </parent>
-    <groupId>com.redhat.lightblue.mongo</groupId>
     <artifactId>lightblue-mongo-hystrix</artifactId>
     <packaging>jar</packaging>
-    <version>1.5.0-SNAPSHOT</version>
```
Comment (jewzaam) [inline hystrix/pom.xml]: Why is groupId and version removed?

File: pom.xml (hunk 1 above)
Comment (dcrissman) [inline pom.xml]: I was unaware of this pattern, but that said, the dependency isn't realized until a child module defines it in its dependencies.

Comment (bserdar) [inline pom.xml]: This ties the version of the referenced project to the version of the root pom. Those two things should not be tied together. Allows for creating a dependency in a sub-project to a version of another entity whose version number is determined by the root pom version number. Please remove this.

File: hystrix/pom.xml (hunk 2 above)
Comment (dcrissman) [inline hystrix/pom.xml]: These are provided via the parent pom and maven considers them to be redundant.

File: pom.xml (hunk 1 above)
Comment (dcrissman) [inline pom.xml]: Reverted in #117

File: hystrix/pom.xml (hunk 2 above)
Comment (jewzaam) [inline hystrix/pom.xml]: This assumes versions are consistent within a repo.  If that assumption is made then #117 shouldn't have been necessary.  I would like this to be added back in for two reasons: 1) it's explicit, 2) we know it works with the release scripts.  If we decide that all versions will be identical in all modules of a repo then we could remove this assuming the release script works, but I don't think we've explicitly had that discussion yet.

Comment (dcrissman) [inline hystrix/pom.xml]: Reverted in #118

**Pull request merged**